### PR TITLE
To Do list: add & remove.

### PR DESCRIPTION
**Pull request by sam2if**
In this project:
I have added the file add.js in the module directory in order to add tasks, remove tasks and edit tasks.

### CODE REVIEW

File: src/module/add.js [hunk 1]
```diff
@@ -1,29 +1,69 @@
 const desc = document.querySelector('.inp');
-const el = document.createElement('div');
 const lists = document.querySelector('.lists');
 
 const tasks = JSON.parse(localStorage.getItem('tasks')) || [];
 
 class Task {
-  constructor(desc, complete) {
+  constructor(desc) {
```
Comment (tufoinnkuo10) [inline src/module/add.js]: - [ ] Great job so far, everything is going well. Please, you are missing the property completed and index value. as mentioned in the guidelines, By default, new tasks should have the property `completed` set to false and the property `index` set to the value of the new array length and Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. please kindly add and fix that. nice job.
- [optional] please ensure it is not possible to add an empty task(with no text) 


Screenshot (946)_LI

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @sam2if 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ CRUD methods well implemented

Required Changes ♻️
Check the comments under the review
- [required] please your MIT link is broken. kindly fix it. we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/module/add.js [hunk 2]
```diff
@@ -1,29 +1,75 @@
...
 const tasks = JSON.parse(localStorage.getItem('tasks')) || [];
 
 class Task {
-  constructor(desc, complete) {
+  constructor(desc, complete, index) {
     this.description = desc.value;
     this.complete = false;
+    this.index = tasks.length;
   }
 }
```
Comment (Owusu-Desmond) [inline src/module/add.js]: - [ ] Please make sure that the property `index` is set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal 5). :blush:  But currently, your app does not work in the same way as you can see so make sure your fix it. I think you achieve that by making `this.index = tasks.length + 1` :+1: 
  
  [Screencast from 04-07-2023 01:39:57 AM.webm](https://user-images.githubusercontent.com/92303891/230523202-1a293b4c-7bc6-4199-9ea3-8bfbe18ce9fe.webm)

- [ ] Also, make sure that deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1) :pray:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @sam2if , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by didierganthier — APPROVED** [review]
Hi @sam2if ,

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀



## Highlights
- Descriptive PR ✔️ 
- Linters passing ✔️ 
- Correct GitHub flow ✔️ 

## Optional Suggestions 💡
- [ ] Well done overall 👏🏽 my only suggestion to improve the project would be to break the js code into more modules. 🤓 

Cheers and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Once again, congratulations on completing this project! 🎉

**Pull request merged**